### PR TITLE
wip: tests for new signatures specifications

### PR DESCRIPTION
**Pull request by jansorg**
makes use of yet unused action `new_file` and the new action `remove_text`.
Corresponding quip doc: https://kite.quip.com/NdIbAwg272aJ/Spec-Function-Signatures-Editor-Plugins-UIUX-WIP

The select events are not yet handled, i.e. the 2nd part of the spec

### CODE REVIEW

File: tests/new_signature_spec/signature_input_text_single.json [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+    {
+      "description": "moving cursor at end of identifier",
+      "step": "action",
+      "type": "move_cursor",
+      "properties": {
+        "offset": 3
+      }
+    },
+    {
+      "description": "typing open parent '('",
```
Comment (abe33) [inline tests/new_signature_spec/signature_input_text_single.json]: `parent` or `paren`?

File: tests/new_signature_spec/signature_input_text_multiple.json [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+    {
+      "description": "moving cursor at end of identifier",
+      "step": "action",
+      "type": "move_cursor",
+      "properties": {
+        "offset": 4
+      }
+    },
+    {
+      "description": "typing 'bar'",
```
Comment (abe33) [inline tests/new_signature_spec/signature_input_text_multiple.json]: This won't work in VSCode, as I explained yesterday, VSCode will only calls signatures providers when typing a `(` or a `,` (as we specified in the provider registration), when the panel is visible any input will trigger a new request, but given the context (the file was open containing an already existing function call) the panel won't be shown and all following inputs won't trigger signature requests.

File: tests/new_signature_spec/signature_remove_text_multiple.json [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+    ]
+  },
+  "test": [
+    {
+      "description": "opening a file",
+      "step": "action",
+      "type": "new_file",
+      "properties": {
+        "file": "data/whitelisted/file.py",
+        "content": "foo(bar, baz"
```
Comment (abe33) [inline tests/new_signature_spec/signature_remove_text_multiple.json]: Same thing in VSCode, the signature panel being not visible, no input will trigger a request except for a `(` and a `,`.

File: tests/new_signature_spec/signature_input_text_single.json (hunk 1 above)
Comment (jansorg) [reply tests/new_signature_spec/signature_input_text_single.json]: paren, of course

**Comment by jansorg** [comment]
@abe33 I've disabled the new test in `default.json` and enabled it for PyCharm.

**Pull request merged**